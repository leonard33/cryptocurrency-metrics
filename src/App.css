.App {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  border-radius: 10px;
  background-image: url("./assets/stat.jpg");
}
