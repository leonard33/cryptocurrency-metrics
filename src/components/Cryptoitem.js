import React from "react";

const Cryptoitem = () => {
  return <div>Cryptoitem</div>;
};

export default Cryptoitem;
