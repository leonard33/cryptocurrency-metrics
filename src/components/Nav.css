.nav {
  align-items: center;
  justify-content: center;
  display: flex;
  color: blue;
}

.back {
  position: absolute;
  top: 20px;
  left: 10px;
  font-size: 30px;
  color: black;
}
