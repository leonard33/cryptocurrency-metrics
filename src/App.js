import { Routes, Route } from "react-router-dom";
import "./App.css";
import Cryptolist from "./components/Cryptolist";
import Nav from "./components/Nav";

function App() {
  return (
    <div className="App">
      <Nav />
      <Routes>
        <Route path="/" element={<Cryptolist />} />
      </Routes>
    </div>
  );
}

export default App;
