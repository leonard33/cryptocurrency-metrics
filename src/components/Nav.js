import React from "react";
import cryptologo from "../assets/crypt.png";
import { AiOutlineLeft } from "react-icons/ai";
import "./Nav.css";
import { Link } from "react-router-dom";

const Nav = () => {
  return (
    <div className="nav">
      <div className="back">
        <Link to={"/"}>
          <AiOutlineLeft color="white" fontWeight={900} />
        </Link>
      </div>
      <div className="logo">
        <img src={cryptologo} width="70" height="50" />
      </div>
      <div className="logo-name">
        <h1>ryptometrics</h1>
      </div>
    </div>
  );
};

export default Nav;
