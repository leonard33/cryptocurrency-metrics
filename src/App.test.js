import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "../../reduxcaptone/space-travelers-hub/src/redux/configureStore";
import App from "./App";
import Cryptoitem from "./components/Cryptoitem";
import Nav from "./components/Nav";

test("renders learn react link", () => {
  render(
    <BrowserRouter>
      <Nav />
    </BrowserRouter>
  );
  const heading = screen.getByText(/ryptometrics/i);
  expect(heading).toBeInTheDocument();
});

// test snapshot
const MockApp = () => {
  <Provider store={store}>
    <BrowserRouter>
      <App />
    </BrowserRouter>
  </Provider>;
};

it("matches snapshot", () => {
  const tree = renderer.create(<MockApp />).toJSON();
  expect(tree).toMatchSnapshot();
});

// test the cryptoitem card
const cryptoitem = [
  {
    uuid: "Qwsogvtv82FCd",
    symbol: "BTC",
    name: "Bitcoin",
    coinrankingUrl: "https://coinranking.com/coin/Qwsogvtv82FCd+bitcoin-btc",
    marketCap: "388246262375",
  },
];

const MockCryptoitem = () => (
  <Provider store={store}>
    <Cryptoitem missioncard={cryptoitem} />
  </Provider>
);

test("renders data in the item as from API", () => {
  const tree = renderer.create(<MockCryptoitem />).toJSON();
  expect(tree).toMatchSnapshot();
});
