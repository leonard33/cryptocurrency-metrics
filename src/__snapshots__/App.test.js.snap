// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `null`;

exports[`renders data in the card as from API 1`] = `
<div
  className="cryptoitem-container"
>
  <div
    className="icon-container"
  >
    <div
      className="icon"
    />
    <div
      className="symbol"
    />
  </div>
  <div
    className="left"
  >
    <div
      className="coin-name"
    />
    <div
      className="marketcap"
    >
      +
    </div>
  </div>
</div>
`;

exports[`renders data in the item as from API 1`] = `
<div
  className="cryptoitem-container"
>
  <div
    className="icon-container"
  >
    <div
      className="icon"
    />
    <div
      className="symbol"
    />
  </div>
  <div
    className="left"
  >
    <div
      className="coin-name"
    />
    <div
      className="marketcap"
    >
      +
    </div>
  </div>
</div>
`;
